crates/aptos-rosetta/src/main.rs

```diff
@@ -13,15 +14,58 @@ use std::{
         atomic::{AtomicBool, Ordering},
         Arc,
     },
+    thread,
+    time::Duration,
 };
+use tokio::time::Instant;
+
+const REST_API_WAIT_DURATION_MS: u64 = 60;
```

Comment:
These params combined mean we're gonna retry 5000 times in the worst case, which feels like a lot. Maybe increase the delay between retries?

Beyond that, perhaps using something like https://docs.rs/again/latest/again/ might be a good call, then we can do retries with jitter, exponential backoff, etc. At the least it helps do this for you.